pagarme/react-scripts-former-kit-dashboard
# Update dependencies

## Context
- Update dependencies
- Refactor lint errors in components

## How to test

1. Install the [create-react-app](https://github.com/facebook/create-react-app) package in the global scope
```
npm install -g create-react-app
```
2. Clone this project
```
git clone -b update/dependencies git@github.com:pagarme/react-scripts-former-kit-dashboard.git
```
3. Run the `create-react-app` with this project as recipe(`--script-version`)
```
create-react-app --scripts-version=file:/path_to_react-scripts-former-kit-dashboard  yourAppName
```

#### Test with an existing project
1. Clone this project
```
git clone -b update/dependencies git@github.com:pagarme/react-scripts-former-kit-dashboard.git
```
2. Go to the project folder
```
  cd react-scripts-former-kit-dashboard/
```
3. Build a pack using the npm pack command
```
npm pack
```
4. In your existing project change the `package.json` refence to `react-scripts-former-kit-dashboard` to the path for the created pack. 
![screen shot 2019-01-16 at 20 00 45](https://user-images.githubusercontent.com/1773766/51281404-71360f80-19c9-11e9-9ea7-5ab20ea21400.png)

5. In your project  folder remove the `node_modules` and reinstall everything
```
  rm -rf node_modules/
  yarn
```
6. Run some command like `yarn lint` or `yarn start`, in the [Pilot project](https://github.com/pagarme/pilot) the `yarn lint` command should trigger a lot of lint errors.

## Checklist
- [x] Update CI config
- [x] Fix lint updates
- [x] Fix former-kit usage
- [x] Add bundle analyzer

## Screenshots


## Deploy notes
**New scripts**:
- `yarn build:analyze` : runs the build process and the bundle analyzer over the build result.

**New dependencies**:

- `cssnano`: used to uglify and minify the css bundle.
- `postcss-safe-parser` : allows a safe css parse for the `cssnano` minification.
- `webpack-bundle-analyzer`: used to analyze the build bundle size and it composition.
